### PR TITLE
BLAC-623 wikipedia citations

### PR DESCRIPTION
**Pull request by CEdwardsBlasikiewicz**
do tests for 'build_via' and 'build_access_date' seem redundant?

### CODE REVIEW

File: app/services/nla/citations/wikimedia_citation_service.rb [hunk 1]
```diff
@@ -106,10 +116,24 @@ def build_language
...
+      def build_persistent_url
+        persistent_url = @document.id
+        if persistent_url.present?
+          " | url=https://nla.gov.au/nla.cat-vn#{persistent_url}\n"
+        end
+      end
+
+      def build_via
+        via = "National Library of Australia"
+        if via.present?
```
Comment (yetti) [inline app/services/nla/citations/wikimedia_citation_service.rb:135]: `via` is always present, so you should just return the string.

File: app/services/nla/citations/wikimedia_citation_service.rb [hunk 2]
```diff
@@ -106,10 +116,24 @@ def build_language
         end
       end
 
-      def build_pi
-        pi = @document.pi
-        if pi.present?
-          " | url=#{pi.first}\n"
+      def build_access_date
+        date = Time.zone.today.strftime("%d %B %Y")
+        if date.present?
```
Comment (yetti) [inline app/services/nla/citations/wikimedia_citation_service.rb:121]: `date` is always present, so just return the string.

File: spec/services/nla/citations/wikimedia_citation_service_spec.rb [hunk 3]
```diff
@@ -121,22 +121,40 @@
...
-        expect(service.build_pi).to eq(" | url=https://nla.gov.au/nla.obj-234175885/flightdiagram\n")
+      it "returns nil" do
+        expect(service.build_persistent_url).to be_nil
       end
     end
+  end
 
-    context "when there is no pi" do
+  describe "#build_access_date" do
+    context "when date is correct" do
```
Comment (yetti) [inline spec/services/nla/citations/wikimedia_citation_service_spec.rb:143]: You should use `freeze_time` or `travel_to` to test against a consistent date or time. https://api.rubyonrails.org/v5.2.4/classes/ActiveSupport/Testing/TimeHelpers.html#method-i-travel_to

**Review by yetti — APPROVED** [review]
You should still include the tests for `via` and the access date. The format might be changed by someone else in the future and it's a simple check to run the tests and make sure what's expected is still there.

File: app/services/nla/citations/wikimedia_citation_service.rb [hunk 4]
```diff
@@ -106,10 +116,24 @@ def build_language
...
+        if persistent_url.present?
+          " | url=https://nla.gov.au/nla.cat-vn#{persistent_url}\n"
+        end
+      end
+
+      def build_via
+        via = "National Library of Australia"
+        if via.present?
+          " | via=#{via}\n"
         end
```
Comment (scott-yeadon) [inline app/services/nla/citations/wikimedia_citation_service.rb:137]: Don't think these functions are needed as the way they are coded the present() method will always be true, two with fixed strings and on with Solr document id which must be set to have existed in Solr. I'd just either return the string or just merge the concatenation with the code above it.

**Review by scott-yeadon — APPROVED** [review]

**Pull request merged**